yapapi/executor/__init__.py

```diff
@@ -416,7 +416,10 @@ async def worker_starter() -> None:
                     and len(workers) < self._conf.max_workers
                     and work_queue.has_unassigned_items()
                 ):
-                    provider_id, b = random.choice(list(offer_buffer.items()))
+                    provider_id, b = max(
+                        random.sample(list(offer_buffer.items()), len(offer_buffer)),
+                        key=lambda el: el[1].score,
+                    )
```

Comment:
The code looks good. But I think the purpose of `random.sample(...)` may be not obvious here, I'd structure this code as follows:
```
shuffled_offers = random.sample(...)
provider_id, offer = max(shuffled_offers, key=...) 
```

Reply:
Restructured as suggested.